### ISSUE TITLE
[🐛 Bug]: Better resolve dynamic imports in browser tests

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#12056]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

latest

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
n/a
```

### What happened?

When running browser tests and importing CJS modules through Vite, it sometimes happen that we get the following error:

```
Error:  Test failed due to following error(s):
  - pr-eact.test.tsx: Failed to fetch dynamically imported module: http://localhost:39259/node_modules/.vite/deps/@testing-library_preact.js?v=60c2fbb0: TypeError: Failed to fetch dynamically imported module: http://localhost:39259/node_modules/.vite/deps/@testing-library_preact.js?v=60c2fbb0
```

Usually it resolves when reloading the page which is why we look for this error in the [browser runner](https://github.com/webdriverio/webdriverio/blob/60adf1fc14c9f4467422769d367b278f14232613/packages/wdio-runner/src/browser.ts#L171). However we have experienced in our build that sometimes even reloading doesn't resolve the issue.

### What is your expected behavior?

The CJS transformation should work flawlessly, the error should not appear.

### How to reproduce the bug.

Unskip [this test](https://github.com/webdriverio/webdriverio/blob/60adf1fc14c9f4467422769d367b278f14232613/e2e/package.json#L9) and run it in CI.

### Relevant log output

```typescript
- pr-eact.test.tsx: Failed to fetch dynamically imported module: http://localhost:39259/node_modules/.vite/deps/@testing-library_preact.js?v=60c2fbb0: TypeError: Failed to fetch dynamically imported module: http://localhost:39259/node_modules/.vite/deps/@testing-library_preact.js?v=60c2fbb0
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by nzakas** [comment]
Looks like our browser tests are passing again.

Because it seems like our tests fail frequently after a new wdio release, would it be possible for you to add an ESLint integration test as part of your release CI?

**Comment by nzakas** [comment]
@christian-bromann thank you 🙏

**Comment by RoXuS** [comment]
hey!

We experiment the same randomly errors, it happens only on gitlab (with a standalone chrome) AND on random tests.
I think it link to the CPU/memory because on our Mac ARM we don't have seen this error.

We use "lit" preset and we are migrating all our 1400 tests to webdriverio.

But we can't continue our migration, our pipelines doesn't pass anymore...
It seems to happen with tests using some external libraries like sinon or fetch-mock.

It started when we added these 2 imports:
```
import fetchMock from 'fetch-mock/esm/client';
import sinon from 'sinon';
```

The error:
```
errors: [
[0-3]     {
[0-3]       filename: '/builds/insight/frontend-monorepo/monorepo/packages/libs/business-components/dt-contract-page/test/dt-contract-page-custom-fields/dt-contract-page-custom-fields.wdio.ts',
[0-3]       message: 'Failed to fetch dynamically imported module: http://localhost:42635/node_modules/.vite/deps/fetch-mock_esm_client.js?v=0d644ae5: TypeError: Failed to fetch dynamically imported module: [http://localhost:42635/node_modules/.vite/deps/fetch-mock_esm_client.js?v=0d644ae5'](http://localhost:42635/node_modules/.vite/deps/fetch-mock_esm_client.js?v=0d644ae5%27)
[0-3]     }
[0-3]   ],
```

**Comment by christian-bromann** [comment]
@RoXuS thanks, this is very interesting. This issue is on top of my list so hopefully I have a solution to this soon to unblock you.

> We use "lit" preset and we are migrating all our 1400 tests to webdriverio.

That's awesome!

**Comment by RoXuS** [comment]
yeah it is really frustrating, to avoid this issue we have to see maxInstances to 1 in case of CI:
```
maxInstances: process.env.CI ? 1 : 6
```

It is ok but execution take 11 minutes instead of 4 minutes on local...

**Comment by piotrblasiak** [comment]
Also experiencing this. setting maxInstances works around it but makes it really slow, of course.
